app/uk/gov/hmrc/tariffclassificationfrontend/controllers/RenderCaseAction.scala

```diff
@@ -47,13 +47,35 @@ trait RenderCaseAction extends FrontendController with I18nSupport {
     }
   }
 
+  protected def defaultRedirect (implicit request: AuthenticatedCaseRequest[_]): Future[Result] = {
+    successful(Redirect(redirect(request.`case`.reference)))
+  }
+  protected def validateAndRedirect(toHtml: Case => Future[Call])
+                                     (implicit request: AuthenticatedCaseRequest[_]): Future[Result] = {
+
+    if (isValidCase(request.`case`)(request)) {
+      toHtml(request.`case`).map(Redirect)
```

Comment:
Is it maybe possible to refactor slightly and create a default implementation for this call along the lines of the defaultRedirect above? There's 3 places in the file where we make a call of 'convert case to html and then map to x'

Reply:
I would agree is was just one parameter, but in this case to refactor would be send a request or Case and the action after the map. Not sure if will look more clear ... I think it looks quite readable now :)